### ISSUE TITLE
xGov Grant Payment Discussion

### ISSUE THREAD

**Issue by SudoWeezy** [algorandfoundation/xGov#57]
As your proposal went through the first step, we would like your opinion on how payment will be handled for the xGov Process. 
For the current session, when a proposal is approved in the xGov portal: 
1 - No Algo upfront and another vote before releasing the Payment if the project is delivered
2 - Threshold of Algos given up front and a vote to give the remainder

> If option 2, what, in your opinion is an appropriate threshold (Give an estimate in USD/Euro)

 Should those 2 options be available for future sessions when creating a proposal?  

@No-Cash-7970
@LoafPickleWW
@scholtz
@ipaleka
@0xLiquidGlass
@mpetruska 
@papasotiriou
@aquilesdel90
@Vidhyanandcs
@dragmz @kerrilija
@PrasoonPratham
@1m1-github 
@HashMapsData2Value
@gidonkatten
@stefanofa
@palace22
@grzracz
@ErikHasselwander
@BunsanMuchi
@Zykoz
@tinchosig
@solkimicreb
@jack-jackhui
@Bhaney44
@SilentRhetoric

**Comment by mpetruska** [comment]
Hi! I'm perfectly fine with option 1 for my proposal, as that was my original understanding of the process. Not sure if that's appropriate for everyone, maybe there should be a discussion on a case-by-case basis between authors and selected governors?

**Comment by grzracz** [comment]
In my opinion option 2 is good, because it is similar to how the old grant system used to work (which was nice).
For this, perhaps ~10% of the grant amount should be paid initially and the rest either per milestone or upon completion of all milestones (perhaps the latter is better if this is a public vote).
Maybe it would be useful to have an xGov site where milestones can be submitted to the public?

**Comment by 2lobo** [comment]
imo this should be part of the proposal, since depending on the proposals they might need more/less than others upfront. obviously we need a solution for this period but for the coming ones the proposer should say what they need upfront and why they need it

**Comment by SilentRhetoric** [comment]
> Maybe it would be useful to have an xGov site where milestones can be submitted to the public?

If milestone delivery evidence were uploaded to GitHub in a standard way, similarly to the proposals, it would make it easier to for xGovs to review.  Emphasis on "standard way."

Perhaps each proposal gets a folder in the xGov repo that contains the original proposal, then subsequently any completion evidence, etc.

**Comment by SilentRhetoric** [comment]
> imo this should be part of the proposal, since depending on the proposals they might need more/less than others upfront. obviously we need a solution for this period but for the coming ones the proposer should say what they need upfront and why they need it

Agree that the payout schedule should be defined in the proposal.  Some proposals may need 100% up front (CEX listing fee, e.g) while others need only 10% up front while others yet can be paid only when the milestone is delivered.  

That being said, I think the administrative burden on xGovs should be carefully considered.  For example, xGovs should only need to vote one additional time for any given proposal to approve milestone delivery, not 2/3/4 times.

**Comment by ipaleka** [comment]
> xGovs should only need to vote one additional time for any given proposal to approve milestone delivery, not 2/3/4 times.

Multiple questions arise:

- is voting on delivery under the same session as the Voting Session?
  - if yes, that implies the same format for the "delivery voting" session as for the Voting Session
  - if no, the core principle of "voting quarterly" (4 times per year) is compromised and xGovernors' obligations are increased without any added benefit

- for the milestone-based proposals (> 10,000 ALGO), voting for the next milestone implies satisfaction with the delivery of the previous milestone, isn't it?

https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1669216855

**Comment by 0xLiquidGlass** [comment]
Option 2

Threshold: 1500 USD

**Comment by LoafPickleWW** [comment]
Personally, I think having this like a rebate system for option 1 is unfair as there are start up costs.

I think 10kA should be dished out right away to anything that passes since that is the Non-Milestone Quota. For this cycle, since it was not required, projects should then allocate a certain % of the requested funds to each milestone. 

For future xGov cycles, anything over 10kA should have the budgeted rewards built into the proposals and it will be not marked as Final until all of the money requested is allocated in the proposal.

**Comment by SilentRhetoric** [comment]
I propose these two options to keep the process simple to understand, participate in, and administrate:

1. Proposal requests funds up front, and they are paid immediately upon the measure passing.  (Example use case: paying an auditor who requires a lump sum payment in advance)
2. Proposal requests funds after delivery of a **single** milestone with clearly-defined acceptance criteria within the proposal.  This approach requires xGov to somehow vote again to disburse the funds after verifying milestone delivery.

We were directed not to include multiple milestones in each proposal, and I also don't think it is tenable to have one proposal contain 2/3/4+ milestones which xGovs must then verify.  This could create a cottage industry of milestone verification and also lock up funds in the pool while subsequent milestones are still pending.

**Comment by ipaleka** [comment]
> Bigger proposals/milestones should be submitted only after 1st milestone (up to 20k) is finished!

I do agree with your suggestions, tho not with this one. The whole bunch of projects won't be able to pass or they will just *make up* some prerequisites for the proposal just to satisfy the form.

**Comment by ipaleka** [comment]
> We were directed not to include multiple milestones in each proposal, 

This is a misunderstanding, you were directed **to include** multiple milestones in your proposal, but you were directed to set the asking funds for only the first milestone.

> and I also don't think it is tenable to have one proposal contain 2/3/4+ milestones which xGovs must then verify. This could create a cottage industry of milestone verification and also lock up funds in the pool while subsequent milestones are still pending.

No pending, once your milestone isn't voted for (funded) your project is dead gone, simple as that.

**Comment by SilentRhetoric** [comment]
I think we need to decouple two concepts in this discussion to clarify things:

A. Does a proposal's overall **vision** require multiple milestones?  Sure, that is possible, but any single proposal in a given xGov period should only be requesting a single payout, not multiple.

B. What is the size of the requested grant in a single proposal/payable milestone?  

These two things are separate concepts and should not be conflated or depend on one another.  The early guidance to break up requests >10k Algo into multiple milestones makes no sense anymore in light of further guidance.  What if you need to request a large lump sum up front to pay an auditor?

**Comment by SilentRhetoric** [comment]
> No pending whatsoever, once your milestone isn't voted for (funded) your project is dead gone, simple as that.

I want to address this point separately: if someone submits evidence of work completed and it does not get approved (for whatever reason, such as mundane administrative issues or even something more sinister like xGov whales who may want to sabotage a competing project), the project should have a path to remediate the issue and resubmit for payment approval.  One milestone approval vote gone wrong should not be a death sentence for a project.

**Comment by simonbelingar** [comment]
> > Bigger proposals/milestones should be submitted only after 1st milestone (up to 20k) is finished!
> 
> I do agree with your suggestions, tho not with this one. The whole bunch of projects won't be able to pass or they will just _make up_ some prerequisites for the proposal just to satisfy the form.

i see your point however i think it's not a problem really. The first milestone still needs to be approved by xgovs, if someone is trying to pitch "fake/filler" proposal it wont get voted in imo - in cases like this, when someone would be asking for a huge grant in 1st proposal(i.e. for some reason they can't make it into smaller milestones, or it doesn't make sense) - then it's literaly up to xgovs to vote on it or not. but again, if someone needs huge sum up front to even start doing something, then imo they have very slim chances of passing based on the current climate and votes tallied so far in xgov.

**Comment by simonbelingar** [comment]
> > No pending whatsoever, once your milestone isn't voted for (funded) your project is dead gone, simple as that.
> 
> I want to address this point separately: if someone submits evidence of work completed and it does not get approved (for whatever reason, such as mundane administrative issues or even something more sinister like xGov whales who may want to sabotage a competing project), the project should have a path to remediate the issue and resubmit for payment approval. One milestone approval vote gone wrong should not be a death sentence for a project.

i don't see any indication someone couldn't apply again with their "slightly adjsuted proposal". That should be welcome. E.g. you failed to get funding the first time, you learned from it, you improved and you've submitted again - all good in my book, i would most likely vote for this project(provided i see improvements) as dedication/commitment is one of the aspects i value highly when evaluating team/person.

**Comment by solkimicreb** [comment]
I strongly agree with @ipaleka . Let's not overcomplicate the system with bloated votes as it will become unusable. One vote per project per period should be a set-in-stone fact that we have to design things around.

**Comment by simonbelingar** [comment]
> > the main point about bigger milestones being second or latter proposal, is to build up confidence/track record on specific project.
> 
> Let me rephrase what I was referring to: the funds for this quarter are 2M ALGO. If there exists a big player that wants to take all of them based on their project milestone, they won't be able to do that after your suggestion even if all xGovernors would recognize that proposal as an absolute killer app that will patch the holes in this sinking ship...
> 
> P.S. This "sinking ship" phrase is probably too harsh, but the first "hole" appeared exactly 19 months ago and this xGovernors initiative is IMO the very first time that _we_ have started to actually patch the holes instead of just pumping water out.


_The TOTAL funds for Xgov are 2mil algo, there is no extra top-up algos approved as of today, so one need to work with this in mind, i would be totally fine if we didn't spend all of it at once._

but let's say this big player with fix-it-all idea comes in, unless they are voting for themselves(top 5 wallets hold roughly 50% of xgov voting power) i find it impossible any sane person will be willing to vote/trust on giving someone 2mil algos to deliver whatever they propose without proven track record on the actual delivery. This is why VCs want MVP/POC done, before they invest, this shouldn't be any different for xgov proposals asking large sums. First prove you can build what you wanna build(mvp) then we can talk. Imporant thing here is again **track record on the specific project/proposal** not in general.

but again, if someone is unable to split 200k $ ask into few smaller asks i have hard time believing they will deliver anything. and with **multiple voting sessions/period e.g. one every month** i don't see an issue asking for consecutive milestones to be approved - heck it even allows to adjust costs/ask for more money that way. 

Having a 1st milestone as mandatory small ask does however prevent granting tons of money in asdvance with no accountability and no proven traction/delivery on the proposed idea.

Additionally - The said project that wants to take 2mil for their milestone would be able to do so, but only as 2nd milestone with proven deliverable behind them - and would need to convince people to actually grant them 2mil based on the work done so far - i don't think this is unreasonable ask tbh. 

and again i think we are all putting our trust in xgovs to vote for what is good for the ecosystem, so **one vote per milestone** should be all is needed to approve each proposal - however big ask it might be.

**Comment by LoafPickleWW** [comment]
The more I think about it, if the project gets approved, they should get the funding. We are just making things more complicated for no reason. Rugs and failed businesses are going to happen no matter what safeguards you put in place. 

If a project wants to delegate payments to hopefully get a better shot at their proposal passing and not looking like a rug, that could be done as well, but the xGov vote should be the only vote.

**Comment by ipaleka** [comment]
> Additionally - The said project that wants to take 2mil for their milestone would be able to do so, but only as 2nd milestone with proven deliverable behind them

Sorry that I presented that oversimplified example, it was confusing.

Let's look at Folks Finance's proposal where they ask for around half the total funds (1M) for their first milestone. If your suggestion is approved, they would need to spend 3 months just for the xGovernors to confirm they are serious, effectively prolonging their final deadline for those 3 months. And Folks people don't need to bring evidence they are serious about their proposal if you ask me.

**Comment by SilentRhetoric** [comment]
> Folks people don't need to bring evidence they are serious about their proposal if you ask me.

When you ask for an order (or two) of magnitude more funding than other projects, it raises the bar even higher for evidence of delivery, if you ask me.

**Comment by ipaleka** [comment]
My fault again, I'm so sorry, please let's not make this discussion about the actual context and leave it about the form.

**Comment by gidonkatten** [comment]
For Folks and I imagine a lot of other projects, one of the biggest expenses is auditing and that requires an upfront payment. 

The alternative is that you have to complete all deliverables, wait for a vote and only then book an audit. We wouldn't want to release a product that has not been audited so this whole process will delay releases significantly.

**Comment by solkimicreb** [comment]
> When you ask for an order (or two) of magnitude more funding than other projects, it raises the bar even higher for evidence of delivery, if you ask me.

Those projects require a lot more votes though. I think the vote distribution/threshold is pretty cool with xgov. Good job with that one 👏

**Comment by solkimicreb** [comment]
> The more I think about it, if the project gets approved, they should get the funding. We are just making things more complicated for no reason. Rugs and failed businesses are going to happen no matter what safeguards you put in place.

@LoafPickleWW yes, please. 👏  Don't overdo things. I think a single vote and potentially a "report" system should be enough until we talk about much larger funds. We are basically talking about ways of choking the only viable way of getting funds in an ecosystem that is really struggling atm and that was super duper overfunded very recently. The bear is already very harsh on everyone.

**Comment by SilentRhetoric** [comment]
> We are basically talking about ways of choking the only viable way of getting funds in an ecosystem that is really struggling atm and that was super overfunded very recently. It's a difficult transition to process. The bear is already very harsh on everyone.

This. 

Making this process too excruciating for the builders who are still here runs the risk of strangling the ecosystem at a vulnerable juncture.

**Comment by HashMapsData2Value** [comment]
I am perfectly fine with getting my funds after a delay (assuming my proposal passes) and I would prefer if, for the start of xGovs at least, we hold off on upfront payments. The community is still reeling from the previous grants process that saw a tragic number of entities get a lot of Algo without delivering anything.


Long term solution: if anyone needs funds immediately, perhaps a system could be setup where they can take a loan from the community. E.g. company A apply for 100k Algo and ask that the funds be sent to smart contract S. In the meantime interested members of the community could deposit 95k Algo into S smart contract and make that amount immediately available to A. Once they deliver what was asked for to the community the 100k funds are released to S and the community can withdraw all of it, for an interest of 5.2% (over 3 months).

If A fails to deliver what they promised (within a grace period) the funds don't get released and the members lost their contribution. A loses all credibility.

If A is very well known to the community the interest can be lower. If they're not as trusted they'll need to offer higher interest rates.

Of course if they're completely unknown, or don't want to give up the grants they'll simply not ask for funds up front.

I think this solution is more dynamic than asking for a portion of the funds up front and having xGovs have to decide on that. Rather than xGovs having to judge the credibility of a developer or team they can instead focus on whether the proposal itself will be beneficial to the ecosystem. Credit check can be done by the ecosystem as a whole, who will get an additional avenue for DeFi.

**Comment by 1m1-github** [comment]
option 2 and with all upfront because that is what milestones are for. the funds are used to survive the time needed to work on the project. that needs to be guaranteed for the duration of a milestone.

**Comment by LoafPickleWW** [comment]
These proposals should really be for what can be completed within a 3 month period anyways. If someone can't deliver on what they promised for the first proposal period, we don't vote for them for the next one.

Simple. Easy. Effective.

Release the funds.

**Comment by SilentRhetoric** [comment]
> Long term solution: if anyone needs funds immediately, perhaps a system could be setup where they can take a loan from the community. E.g:

With all due respect, this is pie-in-the-sky stuff and we need an approach that can be implemented a month ago.  We're now well into voting and no one knows how or when they're going to get funded.

We need to focus on getting basic logistics nailed down before we start talking about designing new lending protocols and the like.  

Crawl, walk, run.

**Comment by HashMapsData2Value** [comment]
> > perhaps a system could be setup where they can take a loan from the community. E.g:
> 
> With all due respect, this is pie-in-the-sky stuff and we need an approach that can be implemented a month ago. We're now well into voting and no one knows how or when they're going to get funded.
> 
> We need to focus on getting basic logistics nailed down before we start talking about designing new lending protocols and the like.
> 
> Crawl, walk, run.

I agree that this should have been considered way back. We were all asked to scope our project/milestone to be deliverable within 3 months. So as soon as a dev/team knows for certain that their proposal is in they should start development. Assuming all goes well and the project delivers what they promised the ones involved with xGov (Adri, maybe Stephane, ?) should have the Algos released. If the community at large believe that a project has failed to delivered what they said then the xGov team will need to make an arbitration.

That is my short term solution to this. 

For the longer term I suggest that lending solution for upfront money, so much funds has already been lost and it has a devastating effect on casual developer morale because they feel "why should I contribute my time and effort to Algorand when these grifters got so much for free and then dumped on retail".

Long term I'm also in favor of the idea that xGovs will have to vote AGAINST a project receiving funding at the end of the xGov period if they failed to deliver. If we assume that projects that have passed the vetting phase are more likely to deliver on their promises than not deliver (otherwise people here would not argue for giving out funds upfront) then this should result in the lesser amount of validation work for xGovs.

**Comment by simonbelingar** [comment]
> > Additionally - The said project that wants to take 2mil for their milestone would be able to do so, but only as 2nd milestone with proven deliverable behind them
> 
> Sorry that I presented that oversimplified example, it was confusing.
> 
> Let's look at Folks Finance's proposal where they ask for around half the total funds (1M) for their first milestone. If your suggestion is approved, they would need to spend 3 months just for the xGovernors to confirm they are serious, effectively prolonging their final deadline for those 3 months. And Folks people don't need to bring evidence they are serious about their proposal if you ask me.

you misunderstand what i am saying - all my suggestions are based on the premise there are multiple voting sessions per period. e.g. one each month. So they would need to only wait one month max - which is more than reasonable in any develompment cycle. If we're stuck with one voting session every 3 months then xgov program fails before it starts. But i agree that adding extra milestone could be in some cases uneccessary, attaching suggested adjsuted approach below.

tldr:
- **there should only be one vote per proposal whatever sum it is asking for with no "additional votes to determine delivery"**
- **funds should be paid in full up-front upon passing the vote, i mean we are literaly VOTING to give them funds...**
- _anyone smart will imo split their proposal into multiple milestones/proposals to have better chance of reaching the votes neded_
- **there should be at least one voting session per month to allow fresh ideas (or next milestones) to be presented and not wait 3 months before they can do so. i think this is especially crucial as it also prevents havign to vote on 20,30+ proposals in single session...**
- _i would suggest putting a hard cap on max funds allocated to single proposal at 50% of available funds. this is mostly due to the reason top few xgovs hold 50% voting power, and this will get even more pronounced in next xgov period with stacking rewards - so to prevent the top wallets to simply syphoning out all rewards to their own/friends and family proposals. It's also a good practice to not put all your eggs in one basket, if we formalize it now we prevent the uneccessary discussions around that in the future..._

**Comment by LoafPickleWW** [comment]
Why we are even having this conversation is beyond me. 

Is xGov here to award grants or grant awards?

**Comment by scholtz** [comment]
I believe i have stated the timeline for payment in the grant texts:

[xGov-23](https://github.com/algorandfoundation/xGov/blob/main/Proposals/xgov-23.md#milestone-1---project-approved-prepayment) : Milestone 1 is reached when approved - prepayment 50% so that we can pay developers

[xGov-24](https://github.com/algorandfoundation/xGov/blob/main/Proposals/xgov-24.md#additional-information) : We expect to deliver the product before the payment

[xGov-9](https://github.com/algorandfoundation/xGov/blob/main/Proposals/xgov-9.md) We have already solved the 2FA accounts, so we will do just the redisgn. It would be nice to know how much we can spend on developers for this. 

Also please consider projects which has been already in status "passed" to be accepted and they should launch the development ASAP (before end of the xGovs voting).

**Comment by ipaleka** [comment]
> you misunderstand what i am saying - all my suggestions are based on the premise there are multiple voting sessions per period. e.g. one each month

Please understand that if there would be more than 3 voting sessions in this following 10 months then the Algorand blockchain is dead gone, without any doubt.

Almost everything that responsible people have done in the last 19 months has been proven to be wrong and xGovernors platform is the only step that is different (and iirc its origin is from outside...). 

XGovernors approached this 12-months process following some premises and if we change those premises that would be the last nail in the coffin.

For a fuck sake, we're talking about funding that is equal to 0.1% of the funding from the last years' funding per quarter, for a fuck sake **exactly we here in this discussion** are the last bunch of people that care, and for the fuck sake we are trying to fight fraudsters now?!?!

**Comment by SilentRhetoric** [comment]
@SudoWeezy @nonfungibleab 

Could you jump in here?  The original post on this discussion does not even mention full up-front payment as an option, yet there seems to be an expectation among some of the group that this was how things would work.  Adding additional votes 

I appreciate that we are "building the plane as we fly it" to some extent, but this is getting ridiculous.  I am having trouble keeping track of what is "decided" and what is "being discussed."

We need some more clarity to be able to run our businesses.

I would also like to point out that the xGov FAQ on Algorand Foundation's site states:

>>> Similarly, if it is decided that more than 1 voting session per quarter is needed for the xGov program, will that apply for xGovs of xGT1 as well or not? In case of the latter, does it mean that for the first 12 months of xGov, there will be at most 4 voting sessions?

>>> The xGov program is nascent, and changes will occur. This process evolution - and how changes are applied to it - will be discussed and voted on by the community, so the process itself remains fair.


This set a clear expectation that there would be at most 4 xGov votes per year, and a community vote is required to change this process.  

I simply do not see a way to square that with the suggestion now that additional votes be held to verify projects' delivery and approve payments.

**Comment by GhostOfMcAfee** [comment]
Here are my two cents. I am not a dev and am not associated with any proposal, so feel free to discount my views.

1) On a going forward basis after this vote closes, the terms of distribution should be a required, standardized, and prominent part of every xGov proposal. Each proposal should state clearly how much is requested upfront versus upon completion of the respective milestones (if applicable). 

1a) Requests for upfront payments should be accompanied by an explanation of why the the upfront payment is needed. If approved by xGov vote, payment should be governed by those terms. 

1b) If we want to consider maximum allowable up front payment that can be requested, that may be prudent. Proposals that make unjustifiable up front requests that nonetheless fall under that threshold can and should be ridiculed for doing so and therefore should fail in the vote. 

2) Proposals that do not pass in this current vote, to the extent they seek to reapply, should amend their proposals to comply with the above point about payment terms. Further, proposals that are currently in draft or final stage but not part of the current vote should be required to amend their proposals to comply.

3) With respect to upfront payments to the current crop of proposals (if they pass), I believe that there must be some sort of upfront payment. The current slate of proposers were operating on imperfect information and it does not seem like it was clear to them how this would go. It does little good to effectively kill a project approved by the community because some reasonable amount is not provided upfront. 

3a) I also think in the absence of clear information on this, Algorand Foundation should take a flexible approach based on the input of and needs of the projects that pass. This seems like what was envisioned in ARC 34 when it allocated "0.5 months for negotiation and claiming grants" and stated that the "Algorand foundation will initiate contact with the proposal's author to negotiate final terms and conditions." 

3b) However, after this current vote, I think details regarding upfront vs. milestone must be made a part of the proposal process as outlined above and should be heeded. 

4) With respect to approval of milestone/final payments, if this is going to be a matter of voting by the xGovs, then it must find the right blend of speed, care, and not upsetting/overburdening xGovs by changing rules mid-term. On the one hand, we cannot have multiple votes all spaced out over 3 month increments. Devs should have to shelve their projects for months at a time while they wait to be paid through a clunky process. On the other hand, if you introduce something like mandatory monthly voting mid-term (such that xGovs will lose accumulated rewards if they don't vote on montly payment dispersals) you will have a full on revolt. We need a compromise position. Here is my suggestion:

4a) Voting on payment approvals should be made dynamic and structured into a two tiered system: (i) Discretionary; and (ii) Mandatory. 

4b) Promptly after a project provides proof of its milestone/final completion and requests payment, this is placed onto the portal with the links to supporting documentation (e.g. GitHub, functioning site, etc.) These requests for dispersement that are uploaded in the middle of the xGov term are deemed Discretionary. Discretionary votes are clearly identified as such on the portal. Failure to vote on a Discretionary vote _does not_ result in loss of accumulated rewards. If during this discretionary period greater than 50% of the voting power of the term pool votes either Yes or No to the request, then the dispersal request is deemed acted upon.

4c) If any Discretionary vote on a dispersal request does not obtain greater than 50% of the voting power of the term pool either in favor or against, then the request is rolled over into the next 3 month xGov term. At this point, the previously Discretionary vote now becomes Mandatory. Mandatory votes are clearly identified as such in the portal. Failure to vote on a Mandatory vote _does_ result in a loss of accumulated rewards. For this, we may not want to base it on obtaining greater than 50% of the voting power of the term pool, but rather may want to make it a simple majority of stake based on those that actually voted (this prevents action from being held up because a whale does not vote and thus we don't get greater than 50% of the total term pool yet have a majority of stake that actually voted having a position in favor or against).

**Comment by 2lobo** [comment]
agree with @GhostOfMcAfee (1): proposals should make clear if they need a payment upfront and give reasons on why they need it and what it is used for in the future

also we should separate this quarter from the coming ones similar to what @GhostOfMcAfee (3) said: it wasnt clear how the payment would work so just give the money to all proposers if they pass just for this period and lets focus on a solution for the coming ones

wanted to propose something similar to what @GhostOfMcAfee describes in (4): 
(b) should be a bit more structured tho so xGovs know that for example first week of each month they should check for discretionary votes. why not use a simple majority vote in this case? 50% could be tough.

**Comment by nonfungibleab** [comment]
[Summary of xGov Grant Payment Discussion on Github](https://github.com/algorandfoundation/xGov/issues/57)

**Current Round**

Pay in advance, as the rules were not clear


**Following Rounds**

We will modify ARC-34 and the grant template to clarify the requirements for multi-milestone applications and the payment process.

- Large amounts should not be requested as the first milestone to allow time to build trust between the xGovs and the applicant's delivery capabilities.
- If Algo is requested at the start of a milestone, upon contract signature, applicants must specify the reason(s) in their application. xGovs will consider these special needs when approving proposals.
- Grant payment is to be requested once the project has shared the milestone completion with the community and, in particular, the xGovs. To release the payment, the xGovs must have verified completion and voted in favor of the payment via a non-mandatory payment approval session. This session's results are only valid if at least 50% of the stake votes.
- If the xGovs are unsatisfied with the delivery, a project must act on their feedback and make adjustments before requesting payment again, which will trigger another vote.
- If a grant requires multiple milestones to be fully delivered, the applicant must add the full project roadmap and what other milestones will be required to deliver the project on their first application. This will help xGovs consider the full picture and budget for future rounds.

**Voting Sessions**

For Q4/23, we will have one mandatory voting session to select grants and at least one discretionary voting session to approve payments.

***Important note:  as per ARC-34, applications must abide by the [Algorand Foundation's legal disclaimers and exclusions](https://www.algorand.foundation/disclaimers) and work should not commence before the contract has been signed.***

**Long-term Design**

There are also some very good ideas about the long-term design of the program and we are looking forward to discussing these with you. We will organize a retro session on Discord once Voting Session #1 ends to start defining the priorities.

**Comment by SilentRhetoric** [comment]
> To release the payment, the xGovs must have verified completion and voted in favor of the payment via a non-mandatory payment approval session. This session's results are only valid if at least 50% of the stake votes.

What happens if xGovs are lazy and not enough stake votes in the non-mandatory session to approve payment for a delivered milestone?

**Comment by LoafPickleWW** [comment]
There really should only be one vote.

**Comment by ipaleka** [comment]
@nonfungibleab there's an important but unanswered question [at the top of this thread](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1669216855).

Not providing an answer to it and still going on with your decision AFAIC means that you are ready to continue the way you have done for the last 19 months, let's call it "avoiding accountability" for the purpose of this thread. 

So, please, answer that question, let me c/p here:

## Who prepares and confirms the description/abstract text that will be placed next to the proposal number and name (I suppose) that xGovernors see during the "delivery vote" session?

**Comment by SilentRhetoric** [comment]
This is a summary of open questions I am tracking to be discussed and resolved:
1. What is the standard for displaying the text on the xGov voting portal ("above the fold" without needing to navigate elsewhere for more information)?  This question could be broadened to cover the exact parameters of how the xGov portal will display everything so that proposal authors can prepare accordingly.
2. If an additional voting mechanism to approve milestone payment is added but not made mandatory, what rules will govern this process?  If xGovs are lazy and do not vote, for example, how does a project get paid after delivery?  We should not create processes which can get stuck.  
3. If the Foundation does not want any work to start on a proposal until a contract has been signed, then what should be the parameters for proposal milestone dates?  For example, if the xGov voting period was August, and then the contracting process takes until mid-September, does that mean that proposals should have delivery dates 
4. Why is it necessary for work not to start before a contract is signed?  There is precedent from AF for Development Award grants for work that is already completed.  Can xGov proposals also include such work?  Should there be parameters around this, such as only for open-source tools for the community?
5. How do we manage a potential explosion in the number of verification votes?  Imagine an example where 10 proposals are approved in xGov Period 1, and each proposal has 4 milestones spread quarterly over a 1-year period, adding 10 verification votes per quarter over the next year.  Then, in xGov Periods 2/3/4 there are 10 more such proposals.  By the time you get to xGov Period 5, there are now 40 verification votes to do alongside regular xGov proposals!  This in one reason that I believe delivery verification votes are untenable.  Additionally, the xGov tooling will necessarily become much more complex to help manage all of this, which is problematic in itself.
6. How can we explicitly (including with on-chain mechanisms) link together a series of proposals/milestones related to the same project?  I think this is a critical mechanism to increase accountability for project owners and transparency for xGovernors evaluating proposals in basically any voting system we design.

The more I think about all this, the more I think all grants should be paid after just a single vote passing the proposal, with AF owning the administration of whatever xGovs approve.  Pushing delivery verification to a voting process would be akin to having a legislative body to perform administrative functions that are generally left to an agency in the executive branch of government.

It should be up to proposal authors to explain why the funds should be awarded, either for up-front expenses or because substantive work has already been delivered.  Similarly, xGovs should consider proposals with this in mind and vote accordingly based on the reasons provided for funding, the team's delivery record (in xGov or otherwise), and any other factors.

**Comment by nonfungibleab** [comment]
Creating the xGov Program will always be coming to a community consensus. Part of my role is facilitating and promoting community discussion to collaborate in the **making of** the required processes.

(a) My understanding of the initial process based on the ARC discussions earlier this year was:
- community submits grants requests on github following criteria and disclaimers summarized on ARC-34
- if a grant requires a large sum for multiple milestones, each milestone must be requested as a separate grant;
- AF provides a mechanism for xgovs to approve the grant application;
- when a grant is approved, AF organizes the contract (ensuring that the application doesn't fall outside of the disclaimers specified on ARC 34) and releases the grant payment. 

The current discussion about the payment-approval voting session is based on our online conversations and the dissonance between my understanding of the process (as described above) and the xgovs expectations of only paying out milestones on delivery which created the need for an approval process, as AF is committed to considering xgovs input all the way. This is your program, after all. 

**How I envisioned the first couple of rounds:**
- xGovs would approve grant and/or first milestone requests within a responsible threshold.
- AF would pay the grant approved once the contract had been signed.
- If a grantee didn't deliver, further milestones would not be approved and the program would lose the initial amount.
**Note that discussions earlier in the year didn't lead to a cap in the amount that could be requested, trusting that the xGovs would make good decisions.

(b) On delivery:
- A single milestone grantee would communicate the delivery to the community and answer questions, request reviews, the community would use the product, etc.
- A multiple milestone grantee would communicate their milestone delivery and provide proof alongside the grant request for their subsequential milestone(s) proposal(s). Creating a cascading effect of delivery and trust.

(c) This initial process would be evaluated to create better tools, including:
- phase 1 - an all-in-one grant submission tool that allows for the submission, community discussion, and approval
- phase 2 - all of the above plus an added tool to automated the grant payment once approved.

In the end, we have to decide with only two weeks to go until the voting session ends, I'm open to running this first session as described above in (a) and (b) and continuing the improvements discussion to reach phases 1 and 2, described in (c). 

Now that we see the issues and needs unfold, it'll move faster.

--- Contracts and Payments ---

The contract process is simple. 
1. Once the grant is approved, we send over the contract (template contract creation in progress), so the grantee can evaluate and asks questions. 
2. Once the grantee has reviewed and is happy, they send us their details (name, **company name and position (if applicable)**, and address. We will create the contract with their details and add a link to the github submission in the deliverables annex.
3. We will send the contract to be signed via DocuSign.
4. Payment is made on the next available AF payment cycle following the signature (AF has two payment cycles per month, the middle and end of the month).

**Comment by nonfungibleab** [comment]
@SilentRhetoric, that's a good list of pending items, but I'm refraining from addressing issues unrelated to the grants payment here.

**Comment by ipaleka** [comment]
>  the dissonance between my understanding of the process (as described above) the xgovs expectations of only paying out milestones on delivery which created the need for an approval process, as AF is committed to considering xgovs input all the way. This is your program, after all.

First of all, this is a totally false premise: "xgovs expectations" and "xgov inputs" phrases are nonsense, there's no entity called xgovs (xGovernors) that can be humanized. xGovs is a **mechanism** capable of making a decision when faced with a predefined set of rules. Absolutely nothing else that entity is capable of, let alone communicate with you.

The truth is that you have communicated with a bunch of **other** entities that may or may not intersect with this entity, but that intersection is completely out of your and our scope.

This intro of mine isn't crucial for this discussion, but it is crucial to understand the process you (Foundation) have created and in which we (developers) have participated.

The only thing that will help us here is the logic itself, and that logic says that we simply have to comprehend the process in the future in order to make a decision.

So, who crafts the text xGovernors will see when they vote on delivery?

If you/Foundation do that, whose/which sources you will be using?

If we/proposers will create them, who will approve them prior to committing to the repo?

**Comment by nonfungibleab** [comment]
@LoafPickleWW 💋 I'm a fan of simple, clean and effective.

@ipaleka I agree with you that xGov is not an entity. What I meant with my comment is that community members express their ideas, and the discussion that follows creates the processes we need to run the program and allow for it to evolve. Thank you for giving me the opportunity to clear that up.

To answer your questions:

>> So, who crafts the text xGovernors will see when they vote on delivery?
At all times, grantees are responsible for all information about their grant proposals and deliverables. Including the text displayed on the voting tool. For this voting session, we extracted the abstract from the github proposals.

>> If you/Foundation do that, whose/which sources you will be using?
Github xgov repo is our source of truth for proposals.

>> If we/proposers will create them, who will approve them prior to committing to the repo?
Could you help us understand why you think the information you create needs to be approved before committing?

**Comment by ipaleka** [comment]
> I'm a fan of simple, clean and effective.

We all are. ;)

> @ipaleka I agree with you that xGov is not an entity. What I meant with my comment is that community members express their ideas, and the discussion that follows creates the processes we need to run the program and allow for it to evolve. Thank you for giving me the opportunity to clear that up.

Thank you for letting me know we are indeed on the same track about this. :)

> To answer your questions:

> > So, who crafts the text xGovernors will see when they vote on delivery?

> At all times, grantees are responsible for all information about their grant proposals and deliverables. Including the text displayed on the voting tool. For this voting session, we extracted the abstract from the github proposals.

I'm aware of that and I absolutely support such a process, let me just remind you that I'm talking here about the **delivery vote**, something that has been introduced for the first time ever by @SudoWeezy in the [initial comment here](https://github.com/algorandfoundation/xGov/issues/57#issue-1840815683) and for what the background has been explained in your [previous comment](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1676788242).

> > If you/Foundation do that, whose/which sources you will be using?
> 
> Github xgov repo is our source of truth for proposals.
>
> > If we/proposers will create them, who will approve them prior to committing to the repo?
>
> Could you help us understand why you think the information you create needs to be approved before committing?

Because humans are able to lie and they especially excel in that if there's no countermeasure.

**Comment by nonfungibleab** [comment]
Yes, @ipaleka I understand we are talking about the **delivery vote** process.

Current version --

For now (version 0), the submission of information would still be on github.  Grantees could to submit a PR to their original PR, including info that can be verified by the community, OR create an issue in this main repo to provide comprehensive info and links. Issues are good IMO, because there can be a dialogue like this :)

> Who approves the request to go on the voting tool? 

I don't think we need an approval process for that. AF will verify the info and links provided. Once in the voting tool, xGovs will be able to verify it themselves.

The voting tool would have the request's summary (which we need to define) and a link to the full info (ie. the github issue), so voters could verify the info before approving/rejecting the request. 

** Important to note ** The voting tool available at xgov.algorand.foundation does not work for yes/no votes right now, only percentages voting. We will need to add yes/no voting to it, which is already built, just not integrated to xgov.

Future version ----
For Version 1, the UI could have an two types of updates that would allow the grantee to add info about the project. 
Simple update - sharing news, asking questions or help testing or any other updates with the community.
Delivery update - required for payment release and including the info about the milestone that was completed with the appropriate proof (code, videos, invoices, audit report etc) and verification links.

**Comment by ShaperOfEntropy** [comment]
I strongly believe that all proposals should require some sort of approval of the delivery by the community before any payments are made, regardless of the requested amount - unless it was explicitly stated in the proposal that a portion of the funds is needed upfront or that it was stated in the proposal itself that a defined set of third party entities have agreed to verify and sign-off on the delivery (in which case the approval of the proposal by xGovs signals that they agree with the delivery verification method and delegate their power to the proposed verifiers). Such a verification committee could consist of well-known community members, experts in the field, etc.

In general, accountability and efficient use of community funds should be the top priority of Algorand (especially if the community learned anything from the old grant scheme).

For the approval of a delivery directly by xGovs (which is the default case if nothing else has been specified), it should be up to the grantee to make it easy and understandable to verify for the xGovs. The other option of xGovs having to "report" projects that do not deliver, is a much less efficient use of time of the community as a whole because there are much more xGovs than project proposers. In the latter system, each (responsible) xGov would be expected to check all the deliverables (which can be quite complicated to understand and verify) and try to convince other xGovs in case of issues. This is similar to the inefficiency we already have now during voting, i.e. the current implied ethos of the xGov program that "projects submit proposals and xGovs need to figure out which are worth supporting" to "projects submit proposals and convince xGovs to support them".
While it creates more work for the proposers to convince xGovs a 2nd time, projects that delivered what was promised should have no issues to gather support again because if xGovs do not approve a valid delivery, good developers would tend to leave/not want to build on Algorand but rather go to other chains that would support them better, which is not in the interest of xGovs (i.e. to do self-harm).
If the approval directly by xGovs were required, subsampling methods could be used - e.g. similarly as in consensus with VRF to randomly construct verification committees.

I also would like to point out that the [comment](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1669858709):
> Bear in mind absolutely the most crucial point here: we are compromising the xGovernors obligations here and the whole system is in danger because they thought their obligations come down to vote 4 times a year, nothing less and nothing more!

is incorrect. The program has been envisioned to change - even the obligations of xGovs. This was one of my initial questions and it was included in the official [Algorand Foundation FAQ](https://www.algorand.foundation/general-faq#05-faq-gov)

**Comment by ipaleka** [comment]
> I also would like to point out that the [comment](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1669858709):
> 
> > Bear in mind absolutely the most crucial point here: we are compromising the xGovernors obligations here and the whole system is in danger because they thought their obligations come down to vote 4 times a year, nothing less and nothing more!
> 
> is incorrect. The program has been envisioned to change - even the obligations of xGovs. This was one of my initial questions and it was included in the official [Algorand Foundation FAQ](https://www.algorand.foundation/general-faq#05-faq-gov)

Let me quote the answer here:

> The xGov program is nascent, and changes will occur. This process evolution - and how changes are applied to it - will be discussed and voted on by the community, so the process itself remains fair.

Let's say we are able to prepare the related question for Governance period #9 **and** that the community decides to add a delivery vote to the pile, that means that the very first opportunity to go with a xGovernors delivery vote session is February 2024.

**Comment by ShaperOfEntropy** [comment]
> Let's say we are able to prepare a related question for Governance period # 9 **and** that the community decides to add a delivery vote to the pile, that means that the very first opportunity to go with a xGovernors delivery vote session is February 2024.

Unfortunately, the process does not work like this. 
If that were the case, Governors would have had to confirm ARCs 33 and 34 already to even get the xGov program started.
That never happened - even though I strongly believe it should have and I have personally been vocal since the beginning that both Governance and xGov programs would first need a formal constitution that would define the basics of the programs and how they can be changed.
*(This should have been ratified already at the initial referendum when the node runners decided to support transfer of Governance power to non-node runners.)*

However, Algorand Foundation has taken the approach of a (really) slow transition of governance into the community hands and has taken the liberty of making unilateral decisions multiple times in the past.
Unfortunately, the majority of the community seems content with such an approach. *(I have suggested both boycotting the Governance and having parallel votes to signal the Foundation the work they are putting into Governance is insufficient but too little seemed to have cared.)*
Therefore, I do not see it as a big issue if the obligation for a delivery vote was similarly decided without an official on-chain vote (even though I am against such an approach).

To add to the point, in G5 measure 2 was approved:
> If this Measure is approved, Algos would be made available for projects proposed by the whole community, upvoted by the xGovs **and approved by General Governors**. 

Clearly the current xGov program is different from what has been approved.
*(I voted against the measure exactly because it was too poorly defined and of too broad interpretations.)*

Also, the response:
> The xGov program is nascent, and changes will occur. This process evolution - and how changes are applied to it - will be discussed and voted on by the community, so the process itself remains fair.

Does not say anything about how exactly the voting on the changes to the program will occur and by whom.
ARCs also do not specify how frequently voting periods will take place.
xGovs that signed up for the program were aware of these shortcomings and decided to participate regardless.

**Comment by ipaleka** [comment]
> Also, the response:

> > The xGov program is nascent, and changes will occur. This process evolution - and how changes are applied to it - will be discussed and voted on by the community, so the process itself remains fair.
> 
> Does not say anything about how exactly the voting on the changes to the program will occur and by whom.

Yes, it says! "Voted by the community" has only one meaning.

I understand you completely and I really appreciate your effort in creating your comment, but you have to understand that this ship is sinking exactly for the reasons you brought there.

So, the sentence "Unfortunately, the process does not work like this." should become "Unfortunately, the process hasn't been done like this.".

I beg you, stop thinking about *them* as some higher authority, **we** are the higher authority in this ecosystem and we demand:

# Simple, clean, effective.

**Comment by ipaleka** [comment]
Let me recap something that is IMO significant here:

@nonfungibleab role has been [well explained](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1676788242):
> Part of my role is facilitating and promoting community discussion to collaborate in the making of the required processes.

Adriana's competence is confirmed by: 

> (a) My understanding of the initial process based on the ARC discussions earlier this year was:
>
> - community submits grants requests on github following criteria and disclaimers summarized on ARC-34
> - if a grant requires a large sum for multiple milestones, each milestone must be requested as a separate grant;
> - AF provides a mechanism for xgovs to approve the grant application;
> - when a grant is approved, AF organizes the contract (ensuring that the application doesn't fall outside of the disclaimers specified on ARC 34) and releases the grant payment.

And then comes a bite of the *fruit of the poisonous tree*:

> The current discussion about the payment-approval voting session is based on our online conversations and the dissonance between my understanding of the process (as described above) and the xgovs expectations of only paying out milestones on delivery which created the need for an approval process, as AF is committed to considering xgovs input all the way. This is your program, after all.

Put aside the fact that the only tangible xGovernors expectations have been to pick us who passed the Voting Session 1 for funding, and focus on the "only paying out milestones on delivery".

Btw, for *small* proposals (single milestone; less than 10K ALGO), and in the case when there's no delivery vote, the voting session is basically voting on donation.

In the case of single voting per milestone (a), if xGovernors decide that a future milestone (second, third, fourth, ...) won't pass my opinion is that they will confirm the same twice in a case of "double voting" (delivery of the previous + funding of the next).

If xGovernors decide that a future milestone (second, third, fourth, ...) should pass my opinion is that they will confirm the same twice in a case of "double voting" (delivery of the previous + funding of the next).

My stance is that if we can comprehend some real-world examples in a tough experiment when a vote pair won't be in unison then there should be "double voting".

If we can't comprehend such an example then the delivery vote session would be redundant.

**Comment by scholtz** [comment]
May I suggest that the core issue with the current Algorand DAO system is the voting system?

Mainly the fact that the voting system is not created to be scaled. For example if algorand software would have to be managed by expert governors, there is no chance right now that those 10 Pull Requests over the day would be governed by this voting system.

The main thing for efficient government system is fracionalized categorized delegation of the voting power. It means that people who trust John in IT will delegate him their voting power, and John can delegate his voting power to winder and ahangsu for example.

If @nonfungibleab  will fix this concept, in the voting plaform we will not have discussion on why expert governors cannot vote more than once per quarter, but usecases such as voting after each milestone within grant proposal will be easily possible.

I think this should be done much better as they have it in Cardano.. Each grant requests passes through the process of proposal, commenting, review, voting, and each milestone goes trhough similar process.. People are paid to do their work, so for each assigned task to comment a proposal they get paid. For reviews of comments mods are paid as well. Would it hurt to allocate 1-5% of assets for non proposals to get the rewards for making xgovs do better decision making?

And please do not think that this is something complicated.. It is very easy to do.. I just believe that AF has bad contracted developers for this

Please make the Algorand Foundation the most transparent and biggest DAO out there and it will have the positive imact on us all..

**Comment by ShaperOfEntropy** [comment]
> Yes, it says! "Voted by the community" has only one meaning.

There are infinitely many possible interpretations of who is part of the community.
Is anyone that holds an Algo, only regular Governors, xGovs, anyone that follows Algorand Foundation Twitter account, only node runners, etc.?
Similarly, it does not specify how the voting is supposed to be done - e.g. on-chain vote through regular Governance, an xGov vote, or a Twitter or Reddit pool or any other of the million possibilities.

That is why I commented that the whole process is too poorly defined and of too broad interpretations.

> I beg you, stop thinking about them as some higher authority, we are the higher authority 

Again, the process does not work like that.
Algorand is POS, the higher the stake, the higher the power.
As long as Algorand Foundation holds the "uncirculated" Algo and the majority of stake does not have an issue with them, it is essentially a "trust me bro" system that they will distribute the uncirculated Algo for the good of the community through the programs that they come up with.

Also, to point out for any that might be unaware of the [T&C of Algorand Foundation](https://www.algorand.foundation/disclaimers)
> The Foundation may, in its sole discretion, demand at any time all or any part of the ALGO Token Incentives distributed under the relevant terms and conditions of the Foundation Programs to be repaid by you forthwith

Of course it would be difficult (bordering on infeasible) to enforce such a demand.
But since Algorand Foundation distributed basically all of the tokens as incentives of one of the other kind, every Algo in circulation is subjected to these terms - regardless of whether users are aware of them or not.
That is why it is imprudent to claim that Algorand Foundation is not the higher authority.

Despite everything, I do hope that this will change in the future.
But like it or not, they have the high ground.

**Comment by ipaleka** [comment]
> > Yes, it says! "Voted by the community" has only one meaning.
> 
> There are infinitely many possible interpretations of who is part of the community. Is anyone that holds an Algo, only regular Governors, xGovs, anyone that follows Algorand Foundation Twitter account, only node runners, etc.? Similarly, it does not specify how the voting is supposed to be done - e.g. on-chain vote through regular Governance, an xGov vote, or a Twitter or Reddit pool or any other of the million possibilities.

The only thing that we don't need right now is to dig a few more holes in the ship.

Your paragraph would have made sense if there have been no seven conducted (and the eighth in progress) **community votes** so far in the ecosystem.

Cheers!

**Comment by ipaleka** [comment]
> That is why it is imprudent to claim that Algorand Foundation is not the higher authority.

You misunderstood me about this, I don't want to fill people with such a claim, I just want **you** to for a moment stop **thinking** the way you have done and act as if we developers here in this thread are some superheroes capable of anything.

And by anything, I, first of all, think on the very initial comment here with two Any Color That We Want So Long As It Is Black "options" provided.

My stance is that we'd be capable do that [by plain logic](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1678800866), so please, can you comprehend **ANY** example for which xGovernors are likely to vote for the same proposals in a two opposite ways?

**Comment by SilentRhetoric** [comment]
At this point I think we need clarity of process before any future xGov sessions are conducted.  Way to achieve more clarity could include:

- An ARC could be finalized
- A Trello board could visualize matters which are considered decided vs still in discussion
- The website QA could be updated and expanded
- A full governance vote could codify parameters 

These processes are intricate systems with many players, incentives, tradeoffs, and unintended consequences, so I think it would be prudent to hammer this out further before subjecting either xGovs or proposers to more uncertainty.  If it takes three months to do it (or longer), I would prefer a clear and robust but delayed process over stumbling into another voting period with so much feedback and confusion still swirling around.

**Comment by ShaperOfEntropy** [comment]
> but I refuse to further discuss that

If you do not wish to discuss this matter further, fine by me since it is starting to digress.

Just to explicitly clarify that my responses and provided justifications were to your original claim that the whole system is in danger if additional voting periods were added for the approval of deliverables, and my claims that there exist precedents (unfortunately) for similar changes, with which the majority of stake did not disagree (and should have been/was prepared for them). Therefore, I do not see why it would be any different this time.

> And then this thread comes out with two options, both having additional votes?! Bear in mind absolutely the most crucial point here: we are compromising the xGovernors obligations here and the whole system is in danger because they thought their obligations come down to vote 4 times a year, nothing less and nothing more!

**Comment by ShaperOfEntropy** [comment]
> I, first of all, think on the very initial comment here with two Any Color That We Want So Long As It Is Black "options" provided.

I would say those were just some (perhaps incompetently-worded) initial options to get the discussion going.
Based on the further responses from @nonfungibleab, it seems to me the Foundation is willing to consider other options as well.

> by plain logic, so please, can you comprehend ANY example for which xGovernors are likely to vote for the same proposal in two opposite ways?

I am unsure I understood the examples from your comment correctly.

Nevertheless, I can give an example of when xGovs would vote differently on a proposal for approval of a delivery and an approval for a subsequent milestone.
The simplest scenario is that the project did deliver what was promised in the first milestone, and xGovs acknowledge this by approving the delivery.
However, they do not see that the second milestone is needed/appropriate at the time/as urgent based on the available program funds and other proposals, or that the second milestone has changed since the proposal for the first milestone has been approved, and thus decide not to approve the second milestone.

**Comment by ipaleka** [comment]
> Just to explicitly clarify that my responses and provided justifications were to your original claim that the whole system is in danger if additional voting periods were added for the approval of deliverables, and my claims that there exist precedents (unfortunately) for similar changes, with which the majority of stake did not disagree (and should have been/was prepared for them). Therefore, I do not see why it would be any different this time.

I have to disagree as I see this xGovernors platform as the only way to save our blockchain from a disaster, so for me, a wound in the xGov body will lead to death.

Among other news, I'll just mention one from today - [AlgoScan is gone](<img src="https://github.com/asastats/channel/assets/49662536/10945d56-c2e3-439f-9375-ca77aefce5c5" width="200" />). AFAIC we are one step from the end and that has been Urtho's willingness to stay.

> I am unsure I understood the examples from your comment correctly.

You did, thank you! :)

> Nevertheless, I can give an example of when xGovs would vote differently on a proposal for approval of a delivery and an approval for a subsequent milestone.
> The simplest scenario is that the project did deliver what was promised in the first milestone, and xGovs acknowledge this by approving the delivery.
> However, they do not see that the second milestone is needed/appropriate at the time/as urgent based on the available program funds and other proposals, or that the second milestone has changed since the proposal for the first milestone has been approved, and thus decide not to approve the second milestone.

Let me analyze this example for the first outcome. At the time when the proposal is announced, when the very first pull request is created, officials from the Foundation and/or we (the community) should suggest to the proposer that their proposal should be separated into at least two different proposals in order to avoid the scenario you just mentioned. If the proposal's second milestone doesn't depend on the first milestone or in the case when the first milestone brings some kind of a minimum viable product that xGovernors can be satisfied with, then the obvious approach is to go with multiple proposals.

And for the latter outcome, I believe the reason why proposers have to include all the milestones in the first pull request is exactly to avoid the problem you emphasized. If a subsequent milestone is dependable on one or more previous milestones, then xGovernors should be aware of that from the very start! Voting against the next milestone's funding is basically a synonym for being unsatisfied with the delivery of the previous.

Am I missing something with this analysis?

**Comment by ShaperOfEntropy** [comment]
> Voting against the next milestone's funding is basically a synonym for being unsatisfied with the delivery of the previous.

No, it is not.
While being unsatisfied with the delivery of a previous milestone does (highly likely) imply that one would vote against the next milestone, the other way around is not necessarily true.
For example, limited funds allocated to the xGov program, unforeseen changes in the regulatory environment, raise of more promising competitors, etc. could all lead an xGov to reconsider supporting subsequent milestones even if delivery of previous one might be satisfactory.
It could also be that the proposers themselves have learned that the initially envisioned second milestone is not appropriate anymore and want to pivot.
The xGovs might agree with the fact that a change is necessary but might not agree that the pivot is in the right direction, and thus would not want to support the project further.

**Comment by ipaleka** [comment]
> > Voting against the next milestone's funding is basically a synonym for being unsatisfied with the delivery of the previous.
> 
> No, it is not. While being unsatisfied with the delivery of a previous milestone does (highly likely) imply that one would vote against the next milestone, the other way around is not necessarily true. 

Let me check if we are on the same track here. I created two paragraphs in my comment. The first one analyzes your first example ("they do not see that the second milestone is needed/appropriate at the time/as urgent based on the available program funds and other proposals"), while the second paragraph refers to your second example ("or that the second milestone has changed since the proposal for the first milestone has been approved").

So my sentence you quoted is just a punchline of **that paragraph**:
> And for the latter outcome, I believe the reason why proposers have to include all the milestones in the first pull request is exactly to avoid the problem you emphasized. If a subsequent milestone is dependable on one or more previous milestones, then xGovernors should be aware of that from the very start! Voting against the next milestone's funding is basically a synonym for being unsatisfied with the delivery of the previous.

**Comment by scholtz** [comment]
> > > Voting against the next milestone's funding is basically a synonym for being unsatisfied with the delivery of the previous.
> > 
> > 
> > No, it is not. While being unsatisfied with the delivery of a previous milestone does (highly likely) imply that one would vote against the next milestone, the other way around is not necessarily true.

In my opinion voting against the milestone MUST mean that the milestone has not been reached and payment will not be released and further actions must be done to approve that milestone again. But the voting on milestones must be introduced by the foundation first in some better xGov system. Some projects can have milestone once per month, others twice per 3 months. Also if someone did not get approved milestone, but they fix the reason why they did not get it approved, they must receive the funds asap after new voting, not in 3 months.

I dont get why people think xGovs should vote only once per qarter.. With introduction xGov i was in opinon that algorand wants some experts to participate more in day to day decision making for Algorand DAO.

**Comment by ipaleka** [comment]
> I dont get why people think xGovs should vote only once per qarter..

Iiirc I'm the only one who repeated this multiple times in this thread, so I should explain why.

If it was expected from the xGovernors to vote 52 times per year at the time when they started their XGovernors journey I would have been talking about 13 votes per quarter.

Until a new Governance (or XGovernors') voting session confirms 52 voting sessions per year (or some other quantity), I stand firmly for one vote per quarter.

**Comment by scholtz** [comment]
@ipaleka  .. think big.. what needs to happen so that experts can do 1000 or 10000 voting sessions per year? 

we need to setup delegation system in place because noone is expert in everything

we need to drop the rule that if you do not vote you loose the assets you have locked.. 

with 10000 voting session per year people can vote on really small things like code management, the approval of milestones, grants, hiring, investments, or whatever that needs funds or strategic decision

my question is .. why current voting system was designed the way it was? Algodand Foundation does not want the Algorand to be decentralized, or is it just that their suppliers of the system lacks the idealogy of open decentralized organization and cuts the AF ideas to bare minimum for as high costs as possible?

**Comment by ipaleka** [comment]
> my question is .. why current voting system was designed the way it was?

Sry @scholtz, I refuse to discuss that as that's not the topic here. I know you'd say that's the origin, but we are so tight with the time that any off-topic is counterproductive.

We need to define how the proposers will be funded and my stance is that we simply have to decide what to do without additional harm to the fragile system. AF officials came here with some dangerous premises that the delivery vote is ad acta and they allegedly even invested in developing such a system.

Here in this thread for the first time ever in the ecosystem's history, we have got people who have been selected by the community involved in a discussion. Maybe I am wrong, but my opinion is that we developers here are the very last bunch of people who care about the ecosystem **and** are capable of getting back the ecosystem on the right track. 

My stance is that we first need to contemplate and [define if the delivery vote is even meaningless or not](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1678800866) and if it is then we should start crafting the question for the Governance #9 where the community will decide about the delivery voting session.

AFAIC discussion about every other topic here takes us away from the solution.

**Comment by scholtz** [comment]
> We need to define how the proposers will be funded and my stance is that we simply have to decide what to do without additional harm to the fragile system. AF officials came here with some dangerous premises that the delivery vote is ad acta and they allegedly even invested in developing such a system.

I believed that we are discussing long term future of xGov system, and thats why I think we need to change the basis of it

In this voting session I absolutely understand why Adri decided that everyone will get the funds before the development.. It is because some has requested full prepayment, some requested partial prepayment and some are ok to be paid later. They do not have right now system for approval of milestones, so they decided to prepay everyone and track who delivers what they promised.. This is good decision in my opinion in the scope of what they have

**Comment by ipaleka** [comment]
> They do not have right now system for approval of milestones

They do not have the required authority to go against xGovernors outcome of the Voting Session... Self-announcing and conducting a delivery vote session for passed proposals in the Voting Session 1 and future Voting Sessions is a clear disavowal of the XGovernors authority and it can't exist any other interpretation of such an act.

**Comment by scholtz** [comment]
Ok, so your question for general governor would be:

The payment for the xGov grant can be released only after xGovs vote for successful delivery of the product, and product must be finished until the next xGov session? Yes/No

Note, that if we have now voting window for xGovs 1.8. - 31.8., next voting window will be 1.11.-30.11. .. So all grants must be finished until than in order to get some funds. The contracts with AF will be signed I believe in mid Sept, so developers have from 15.9. to 1.11. approx 70 days to deliver the product or they miss the payment window. (next is 1.2. to 31.2.)

Also AF is very slow in payments, so I assume it will be done at the end of the month following the xGov vote.. 

So if you want to create a product, lets say Open source AMM with concentrated liquditity, and you want to ask for a grant to do so, you have costs to write up the idea and deliver it to xGov github in june, write white paper, persuade all xGovs to vote for it, then you have to start the company so that everything is legal according to MiCA regulations, you have costs for development, legal costs, infrastructure costs, and if you manage to do everything in 70 days and you persuade the expert governors that you delivered the product, you may get funds on christmas..

So from may to christmas you are short on any money to be received, you risk that you do not get the money, you risk that your idea will be stolen by other AMMs who received millions of dollars by AF before and should be happy with xGov system?

In my opinion we need better xGov system, where grant requesters may request money to be paid in front with good reasoning, where all grants are reviewed and reviews are reviewed, where xGovs can delegate their voting power to others, where is no harm if I delegate my voting power to someone I trust he makes good decisions and do not vote on specific voting session, where each milestone has to be reviewed, where those reviewes are also reviewed, where everybody who makes some work is paid for their work, where grants are paid only after the xGov approval, and where xGov is not just for the grants but whole Algorand decision making such as strategic investments, marketing or consensus changes in the protocol.

**Comment by ipaleka** [comment]
> Ok, so your question for general governor would be:
> 
> The payment for the xGov grant can be released only after xGovs vote for successful delivery of the product, and product must be finished until the next xGov session? Yes/No

Nope. 

Nope.

I don't comprehend any question at this moment as AFAIC we have to [define meaningfulness of the delivery vote session](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1678800866) ([#2](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1679679884) [#3](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1679722338) [#4](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1679999808) [#5](https://github.com/algorandfoundation/xGov/issues/57#issuecomment-1680374798)).

> In my opinion we need better xGov system

We probably do. But stretching the discussion in that direction here and now is counterproductive.

**Comment by solkimicreb** [comment]
I think it's safe to say that anything that complicates xGov is a no-go. We have pretty hard evidence.

**Comment by GhostOfMcAfee** [comment]
> What happens if xGovs are lazy and not enough stake votes in the non-mandatory session to approve payment for a delivered milestone?  There would need to be either a) a mechanism to ensure xGovs vote in these sessions and/or b) a way for a completed project to push past xGov inaction and receive payment.

We could make it be >50% **_of those that actually voted_** instead of 50%+ of stake. In that case, lazy xGovs do not hold things up.

**Comment by ipaleka** [comment]
@nonfungibleab 

> Congratulations! xGov voting session 1 is now closed and your proposal has passed. I need your email to kick off contract signature next week. Would you please send me a DM on Twitter or Discord?

<img src="https://github.com/algorandfoundation/xGov/assets/49662536/e87aca3a-f849-4aa4-a3dc-ce4bd11cef56" width=400>
<img src="https://github.com/algorandfoundation/xGov/assets/49662536/31d67402-f1d4-4f82-aa96-34d779c02b26" width=400>

**Comment by PrasoonPratham** [comment]
Hey! We submitted a proposal for an SVM L2 on Algorand that's been approved.

We would like to move forward with the project. 

Could you DM me [here](https://twitter.com/PrasoonPratham) to continue the conversation?

**Comment by nonfungibleab** [comment]
@ipaleka changed my discord settings. please try again.

**Comment by ipaleka** [comment]
> @ipaleka changed my discord settings. please try again.

Nope.

<img src="https://github.com/algorandfoundation/xGov/assets/49662536/d8d813d0-8286-46af-aaaf-f9b7ea9c327c" width="400">
<img src="https://github.com/algorandfoundation/xGov/assets/49662536/0f31840a-4066-42bc-a77c-a073fbf060ee" width="400">

**Comment by ipaleka** [comment]
@nonfungibleab Please initiate removing/updating the following part from the [README](https://github.com/algorandfoundation/xGov#changing-your-proposal-status-ahead-of-a-voting-session) as we won't have any proposal for Session 2 if that applies.

> To ensure your proposal is included in the quarter's voting round, make sure that your proposal status is set to "Final" no later than two weeks after the start of the governance period.

**Comment by SudoWeezy** [comment]
thanks @ipaleka, I updated the Read Me
